core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -403,8 +403,12 @@ private[log] class Cleaner(val id: Int,
    * @param retainDeletes Should delete tombstones be retained while cleaning this segment
    *
    */
-  private[log] def cleanInto(topicAndPartition: TopicAndPartition, source: LogSegment,
-                             dest: LogSegment, map: OffsetMap, retainDeletes: Boolean) {
+  private[log] def cleanInto(topicAndPartition: TopicAndPartition,
+                             source: LogSegment,
+                             dest: LogSegment,
+                             map: OffsetMap,
+                             retainDeletes: Boolean,
+                             messageFormatVersion: Int) {
```

Comment:
Hi Jun, this patch is using the message format version configured for this topic when doing compaction. The configuration is passed in in line 373. Do you mean something else?